### PR TITLE
[api] Set 200 OK response for undelete action on project

### PR DESCRIPTION
**Pull request by saraycp**
This way, the endpoint will render a `200 OK` response instead of a `204 No Content` response when the 'undelete' is correctly performed.

### CODE REVIEW

**Comment by adrianschroeter** [comment]
thanks! but it is an api change, please prefix this with [api] thanks :)

**Review by krauselukas — APPROVED** [review]

File: src/api/test/functional/source_controller_test.rb [hunk 1]
```diff
@@ -2009,7 +2009,7 @@ def test_remove_and_undelete_operations
 
     login_king
     post '/source/kde4', params: { cmd: :undelete }
-    assert_response :success
+    assert_response 200
```
Comment (adrianschroeter) [inline src/api/test/functional/source_controller_test.rb:2012]: just wondering, why is this needed? :success should be 200, no?

Comment (saraycp) [reply src/api/test/functional/source_controller_test.rb:2012]: The `:success` accept any of the successful responses (200–299).
This test passed when we got `204 No content` code, but we really wanted a 200. So I changed it to 200 to be more specific.

Anyway, it's true that I can just leave the `:success`. Do you prefer that?

Comment (saraycp) [reply src/api/test/functional/source_controller_test.rb]: Well, changed to `:success`.  We know it's due to a 200 and not a 204 code.

**Comment by adrianschroeter** [comment]
On Freitag, 3. Dezember 2021, 13:37:25 CET Saray Cabrera Padrón wrote:
> @saraycp commented on this pull request.
> 
> 
> 
> > @@ -2009,7 +2009,7 @@ def test_remove_and_undelete_operations
>  
>      login_king
>      post '/source/kde4', params: { cmd: :undelete }
> -    assert_response :success
> +    assert_response 200
> 
> Well, changed to `:success`.  We know it's due to a 200 and not a 204 code.

ah, ic.

-- 

Adrian Schroeter ***@***.***>
Build Infrastructure Project Manager

SUSE Software Solutions Germany GmbH, Maxfeldstr. 5, 90409 Nürnberg, Germany
(HRB 36809, AG Nürnberg) Geschäftsführer: Ivo Totev

**Comment by saraycp** [comment]
> ah, ic.

Is this an approval, @adrianschroeter? :smiley_cat:

**Comment by adrianschroeter** [comment]
On Freitag, 3. Dezember 2021, 15:01:10 CET Saray Cabrera Padrón wrote:
> > ah, ic.
> 
> Is this an approval, @adrianschroeter? :smiley_cat: 

both is fine, but I understood the reason now :)

-- 

Adrian Schroeter ***@***.***>
Build Infrastructure Project Manager

SUSE Software Solutions Germany GmbH, Maxfeldstr. 5, 90409 Nürnberg, Germany
(HRB 36809, AG Nürnberg) Geschäftsführer: Ivo Totev

**Comment by saraycp** [comment]
> both is fine, but I understood the reason now :)

Then, I'm going to merge this PR if you agree.
